Framework/Algorithms/src/DeadTimeCorrection.cpp

```diff
@@ -117,21 +119,30 @@ void DeadTimeCorrection::exec() {
   for (int index = 0; index < static_cast<int>(grouped->getNumberHistograms()); ++index) {
     PARALLEL_START_INTERUPT_REGION
     progress.report("Performing the correction for the group at index " + std::to_string(index));
-    const double y = grouped->y(index)[0];
-    double correction = 1. / (1. - tau * y);
-    if (y >= 1. / tau) {
-      g_log.warning() << "Saturation count rate reached for grouped detector at index " << index
-                      << ". Correction will be infinity. Check your tau or input "
-                         "workspace, make sure it is normalised by acquisition time.\n";
-      correction = std::numeric_limits<double>::infinity();
+    auto correction = grouped->readY(index);
+    for (size_t bin = 0; bin < correction.size(); bin++) {
+      auto &y = correction[bin];
+      if (y >= 1.0 / tau) {
+        g_log.warning() << "Saturation count rate reached for grouped detector at index " << index << ", in bin " << bin
+                        << ". Correction will be infinity. Check your tau or input "
+                           "workspace, make sure it is normalised by acquisition time.\n";
+        y = std::numeric_limits<double>::infinity();
+      }
     }
+    std::for_each(correction.begin(), correction.end(), [&tau](double &y) { y = 1.0 / (1.0 - y * tau); });
+
     const auto detIDs = grouped->getSpectrum(index).getDetectorIDs();
     for (const auto id : detIDs) {
       const size_t originalIndex = map.at(id);
       auto &spectrum = outputWorkspace->mutableY(originalIndex);
       auto &errors = outputWorkspace->mutableE(originalIndex);
-      spectrum *= correction;
-      errors *= correction;
+      if (correction.size() == 1) {
```

Comment:
This check seems redundant, can you not always do `*=correction` ?

Reply:
That actually leads to an error when correcting TOF. There, the correction has size 1, but the size of spectra can be arbitrary and not equal to the size of correction, which in turn makes it incompatible for multiplying this way.

Reply:
OK, thought ISpectrum would do the broadcasting.